bin/strata-client/src/rpc_server.rs

```diff
@@ -734,3 +797,26 @@ impl StrataSequencerApiServer for SequencerServerImpl {
             .map_err(|e| Error::Other(e.to_string()))?)
     }
 }
+
+pub struct StrataDebugRpcImpl {
+    l2_block_manager: Arc<L2BlockManager>,
+}
+
+impl StrataDebugRpcImpl {
+    pub fn new(l2_block_manager: Arc<L2BlockManager>) -> Self {
+        Self { l2_block_manager }
+    }
+}
+
+#[async_trait]
+impl StrataDebugApiServer for StrataDebugRpcImpl {
+    async fn get_block_by_id(&self, block_id: L2BlockId) -> RpcResult<Option<L2Block>> {
+        let l2_block_manager = self.l2_block_manager.clone();
+        let l2_block = l2_block_manager
+            .get_block_async(&block_id)
+            .await
+            .map_err(Error::Db)?
+            .map(|b| b.block().clone());
```

Comment:
Not sure but maybe we don't need a clone here?

Reply:
.block() returns reference so it's needed in this case.